src/backend/distributed/executor/executor.c

```diff
@@ -1177,8 +1277,14 @@ RunDistributedExecution(DistributedExecution *execution)
 				execution->waitFlagsChanged = false;
 			}
 
-			/* we should always have more (or equal) waitEvents */
-			Assert(maxWaitEventCount >= connectionCount + 2);
+
+			/*
+			 * maxWaitEventCount depends to workerCount in sequential execution
+			 * else to session count.
+			 */
+			Assert(MultiShardConnectionType == SEQUENTIAL_CONNECTION ?
+				   (maxWaitEventCount >= list_length(execution->workerList) + 2) :
+				   (maxWaitEventCount >= connectionCount + 2));
```

Comment:
I feel like this assert is mostly noise.

Reply:
Will remove